### ISSUE TITLE
Does hxnodejs cause the compiler to define the 'sys' define?

### ISSUE THREAD

**Issue by kevinresol** [HaxeFoundation/hxnodejs#59]
Or should it?

**Comment by Simn** [comment]
It does allow the sys package: https://github.com/HaxeFoundation/hxnodejs/blob/master/extraParams.hxml

I don't know if it can actually add `-D sys` because that's a reserved flag. We might have to find a solution for that.

**Comment by kevinresol** [comment]
Yeah but some lib/code depends on the `sys` define. When the sys-related api are complete (e.g. `Sys.stdout()`), I think the compiler should define `sys` for us.

Tried `-D sys` and it complains that I can't set it manually.

**Comment by kevinresol** [comment]
> I don't know if it can actually add -D sys because that's a reserved flag. We might have to find a solution for that.

Is it possible to hardcode inside the compiler, defining `sys` when it sees the `hxnodejs` lib is included?

**Comment by nadako** [comment]
But `hxnodejs` doesn't implement full sys api (because it's impossible without dirty tricks (#64). It just implements some of it, so I don't know if it's reasonable to define `sys`...

**Comment by kevinresol** [comment]
@nadako maybe make a checklist of them and see what we can do.

**Comment by nadako** [comment]
the main problem is that we can't implement synchronous API that are used in sys, like `Sys.stdout()`.

**Comment by kevinresol** [comment]
Will this work? http://stackoverflow.com/a/6471170/3212365

**Comment by nadako** [comment]
Hmm, I don't know. Will have to look into it, not sure if that'll work everywhere.
Thanks for the hint!
